code/database/behavior/identifiable.php

```diff
@@ -90,7 +90,7 @@ protected function _afterSelect(DatabaseContext $context)
     {
         if($this->getMixer() instanceof DatabaseRowInterface && $this->_auto_generate && !$this->isNew())
         {
-            if($this->hasProperty('uuid') && empty($this->uuid))
+            if(empty($this->uuid))
```

Comment:
No the isSupported() method explicitly check if the table has a uuid column. No need to double check it.